### PR TITLE
upgrade deps including Svelte 4

### PR DESCRIPTION
**Pull request by ryanatkn**
Upgrades deps including Svelte 4.

Includes a similar ESLint override hack that I added to the felt-ui PR. We probably just need to wait for a dependency and it'll go away, and linting will fail once it's fixed.

### CODE REVIEW

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -170,7 +160,7 @@
 
 <SocketConnection {socket} url={PUBLIC_WEBSOCKET_URL} />
 
-<Themed {theme}>
+<Themed>
```
Comment (ryanatkn) [inline src/routes/+layout.svelte:163]: the behavior here should be the same

File: src/lib/ui/About.svelte [hunk 2]
```diff
@@ -1,5 +1,8 @@
 <script lang="ts">
 	import {PUBLIC_ADMIN_EMAIL_ADDRESS, PUBLIC_ADMIN_ICON, PUBLIC_GIT_HASH} from '$env/static/public';
+
+	// TODO maybe render this
+	// import {VERSION} from '@sveltejs/kit';
```
Comment (ryanatkn) [inline src/lib/ui/About.svelte:5]: this was just added to SvelteKit, wasn't sure about including it, the string is `'1.21.0'`

File: src/routes/+layout.svelte [hunk 3]
```diff
@@ -162,15 +152,11 @@
 	]}
 />
 
-<svelte:head>
-	<link rel="shortcut icon" href="/favicon.png" />
-</svelte:head>
-
```
Comment (ryanatkn) [inline src/routes/+layout.svelte:168]: noticed this code duplicated what's in `src/app.html`

File: src/lib/ui/About.svelte (hunk 2 above)
Comment (greatbacon) [reply src/lib/ui/About.svelte:5]: ooh, I like that! Maybe even link to the project too?

**Review by greatbacon — APPROVED** [review]
🧹 ✨

File: src/lib/ui/About.svelte (hunk 2 above)
Comment (ryanatkn) [reply src/lib/ui/About.svelte]: sweet yeah added that

**Pull request merged**